qcore/utils.py

```diff
@@ -6,8 +6,149 @@
 from shutil import rmtree
 import os
 import imp
+import yaml
+from collections import OrderedDict
+from collections import Mapping
 
-def setup_dir(directory, empty = False):
+
+class DotDictify(dict):
+    """
+    Construct an dictionary object whose values can also be accessed by 'dot'
+    eg. d.k; d.k1.k2
+    """
+    MARKER = object()
+
+    def __init__(self, value=None):
+        if value is None:
+            pass
+        elif isinstance(value, dict):
+            for key in value:
+                self.__setitem__(key, value[key])
+        else:
+            raise TypeError('expected dict')
+
+    def __setitem__(self, key, value):
+        if isinstance(value, dict) and not isinstance(value, DotDictify):
+            value = DotDictify(value)
+        super(DotDictify, self).__setitem__(key, value)
+
+    def __getitem__(self, key):
+        found = self.get(key, DotDictify.MARKER)
+        if found is DotDictify.MARKER:
+            found = DotDictify()
+            super(DotDictify, self).__setitem__(key, found)
+        return found
+
+    __setattr__, __getattr__ = __setitem__, __getitem__
+
+
+def ordered_load(stream, Loader=yaml.Loader, object_pairs_hook=OrderedDict):
+    """
+    :param stream: yaml file path
+    :param Loader: yaml loader
+    :param object_pairs_hook: always=OrderedDict to load file in order;
+
+    :return: OrderedDict
+    """
+
+    class OrderedLoader(Loader):
```

Comment:
here you are extending the Loader class yet you haven't actually extended the Loader class.